scarlet/initialization.py

```diff
@@ -238,20 +321,48 @@ def init_extended_source(
         )
         morph = prox_monotonic(morph, 0).reshape(morph.shape)
 
-    # truncated at thresh * bg_rms
+    # truncate morph at thresh * bg_rms
     threshold = bg_rms * thresh
-    morph, bbox = trim_morphology(center_index, morph, threshold)
+    morph, bbox = trim_morphology(center_index, morph, bg_thresh=threshold)
     bbox = frame.bbox[0] @ bbox
 
     # normalize to unity at peak pixel for the imposed normalization
     if morph.sum() > 0:
         morph /= morph.max()
+
+        # since the spectrum assumes a point source:
+        # determine the optimal amplitude for matching morph and the model psf
+        # TODO: morph is still convolved with the observed PSF, but we compute
+        # amplitude correction as if it were not..
+        if frame.psf is not None:
+            psf = frame.psf.get_model()
+
+            shape = (psf.shape[0], *morph.shape)
+            bbox_ = Box(
+                shape,
+                origin=(
+                    psf.shape[0] - shape[0],
+                    psf.shape[1] // 2 - shape[1] // 2,
+                    psf.shape[2] // 2 - shape[2] // 2,
+                ),
+            )
+            psf = bbox_.extract_from(psf)
+
+            # if model PSF is constant across bands (as it should) then factor
+            # is constant as well
+            factor = (morph[None, :, :] * psf).sum(axis=(1, 2)) / (psf * psf).sum(
```

Comment:
Redundant code with `get_psf_spectrum`

Reply:
not at all. this isolated equation is formally the same, although it differs in being applied to 3D cubes instead of 2D.  However, many more operations need to be done here.

Just for explanation, `spectrum` assumes the source to have point-source morphology, otherwise `get_pixel_spectrum` has no defined operation. This `factor` corrects that by finding out how much (in terms of a scalar number) `morph` looks like the (model) `psf`.

Leaving code comments here, too